server/middleware/rolloutMiddleware.test.ts

```diff
@@ -0,0 +1,103 @@
+import { Request, Response } from 'express'
+import rolloutMiddleware from './rolloutMiddleware'
+import config from '../config'
+
+const req = {} as Request
+const next = jest.fn()
+
+const middleware = rolloutMiddleware()
+
+beforeEach(() => {
+  jest.resetAllMocks()
+  config.rollout.restricted = true
+  req.url = '/'
+})
+
+describe('rolloutMiddleware', () => {
+  it('should allow any GET to the rollout-status page', async () => {
+    const res = getProbationRes('UNKNOWN')
+    req.method = 'GET'
+    req.url = '/rollout-status'
+    middleware(req, res, next)
+    expect(next).toBeCalledTimes(1)
+    expect(res.redirect).not.toBeCalled()
+  })
+
+  it('should allow GET when a probation user is inside the rollout areas', async () => {
+    const res = getProbationRes('N55')
+    req.method = 'GET'
+    middleware(req, res, next)
+    expect(next).toBeCalledTimes(1)
+    expect(res.redirect).not.toBeCalled()
+  })
+
+  it('should redirect when a probation user is outside the rollout areas', async () => {
+    const res = getProbationRes('N50')
+    req.method = 'GET'
+    middleware(req, res, next)
+    expect(res.redirect).toBeCalledTimes(1)
+    expect(next).not.toBeCalled()
+  })
+
+  it('should allow GET when a prison user is inside the rollout establishments', async () => {
+    const res = getPrisonRes(['MDI'])
+    req.method = 'GET'
+    middleware(req, res, next)
+    expect(next).toBeCalledTimes(1)
+    expect(res.redirect).not.toBeCalled()
+  })
```

Comment:
Might need to alter these tests when we want to set the pilot areas correctly. eg. to Swansea, Cardiff, Usk, Prescoed and Parc.

Reply:
I will probably update the tests to use their own config, overruling what is in the `config.ts` file.